--- requests/models.py.orig	2014-06-07 22:18:49.613062966 +0200
+++ requests/models.py	2014-06-07 22:19:40.417060920 +0200
@@ -158,9 +158,12 @@
     def register_hook(self, event, hook):
         """Properly register a hook."""
 
-        if event not in self.hooks:
+        if event not in self.hooks and event != 'zato_pre_request':
             raise ValueError('Unsupported event specified, with event name "%s"' % (event))
 
+        if event == 'zato_pre_request':
+            self.hooks[event] = []
+
         if isinstance(hook, collections.Callable):
             self.hooks[event].append(hook)
         elif hasattr(hook, '__iter__'):
