--- requests/models.py.orig	2018-06-13 19:01:06.803336282 +0200
+++ requests/models.py	2018-06-13 19:01:38.746338693 +0200
@@ -175,9 +175,12 @@
     def register_hook(self, event, hook):
         """Properly register a hook."""
 
-        if event not in self.hooks:
+        if event not in self.hooks and event != 'zato_pre_request':
             raise ValueError('Unsupported event specified, with event name "%s"' % (event))
 
+        if event == 'zato_pre_request':
+            self.hooks[event] = []
+
         if isinstance(hook, Callable):
             self.hooks[event].append(hook)
         elif hasattr(hook, '__iter__'):
