--- django_openid_auth/models.py.orig	2014-05-07 16:50:38.024290453 +0200
+++ django_openid_auth/models.py	2014-05-07 16:52:26.124286099 +0200
@@ -54,5 +54,5 @@

 class UserOpenID(models.Model):
     user = models.ForeignKey(User)
-    claimed_id = models.TextField(max_length=2047, unique=True)
+    claimed_id = models.CharField(max_length=255, unique=True)
     display_id = models.TextField(max_length=2047)
