--- apscheduler/threadpool.py.orig	2013-12-14 15:25:55.485491410 +0100
+++ apscheduler/threadpool.py	2013-12-14 15:26:34.249683586 +0100
@@ -4,7 +4,7 @@
 ThreadPool!
 """
 
-from threading import Thread, Lock, currentThread
+from threading import Thread, RLock, currentThread
 from weakref import ref
 import logging
 import atexit
@@ -43,7 +43,7 @@
         self.max_threads = max(max_threads, core_threads, 1)
         self.keepalive = keepalive
         self._queue = Queue()
-        self._threads_lock = Lock()
+        self._threads_lock = RLock()
         self._threads = set()
         self._shutdown = False
 
