--- requests/packages/urllib3/connectionpool.py.orig	2015-06-08 18:13:01.136948026 +0200
+++ requests/packages/urllib3/connectionpool.py	2015-06-08 18:13:04.568869138 +0200
@@ -308,12 +308,6 @@
         if hasattr(err, 'errno') and err.errno in _blocking_errnos:
             raise ReadTimeoutError(self, url, "Read timed out. (read timeout=%s)" % timeout_value)
 
-        # Catch possible read timeouts thrown as SSL errors. If not the
-        # case, rethrow the original. We need to do this because of:
-        # http://bugs.python.org/issue10272
-        if 'timed out' in str(err) or 'did not complete (read)' in str(err):  # Python 2.6
-            raise ReadTimeoutError(self, url, "Read timed out. (read timeout=%s)" % timeout_value)
-
     def _make_request(self, conn, method, url, timeout=_Default,
                       **httplib_request_kw):
         """
