--- requests/packages/urllib3connectionpool.py.orig	2014-05-06 18:47:11.346680199 +0200
+++ requests/packages/urllib3connectionpool.py	2014-05-06 18:47:38.246679116 +0200
@@ -134,7 +134,7 @@
     ConnectionCls = HTTPConnection
 
     def __init__(self, host, port=None, strict=False,
-                 timeout=Timeout.DEFAULT_TIMEOUT, maxsize=1, block=False,
+                 timeout=Timeout.DEFAULT_TIMEOUT, maxsize=600, block=False,
                  headers=None, _proxy=None, _proxy_headers=None):
         ConnectionPool.__init__(self, host, port)
         RequestMethods.__init__(self, headers)
@@ -148,14 +148,14 @@
 
         self.timeout = timeout
 
-        self.pool = self.QueueCls(maxsize)
+        self.pool = self.QueueCls(600)
         self.block = block
 
         self.proxy = _proxy
         self.proxy_headers = _proxy_headers or {}
 
         # Fill the queue up so that doing get() on it will block properly
-        for _ in xrange(maxsize):
+        for _ in xrange(600):
             self.pool.put(None)
 
         # These are mostly for testing and debugging purposes.

