--- butler/__init__.py.orig	2014-06-23 02:38:37.928173156 +0200
+++ butler/__init__.py	2014-06-23 02:49:00.564763243 +0200
@@ -31,7 +31,7 @@
         """
         return_obj = self.obj
         for key in path:
-            if type(return_obj) in [list, dict]:
+            if isinstance(return_obj, (list, dict)):
                 return_obj = return_obj[key]
             else:
                 return None
