diff --git a/code/pymqi.py b/code/pymqi.py
index 2d94092..9c32721 100644
--- a/code/pymqi.py
+++ b/code/pymqi.py
@@ -960,6 +960,15 @@ class sco(MQOpts):
         if "6.0" in pymqe.__mqlevels__:
             opts += [['KeyResetCount', 0L, MQLONG_TYPE],
                      ['FipsRequired', 0L, MQLONG_TYPE]]
+            
+        # Stuff new in MQ 7.1
+        if '7.1' in pymqe.__mqlevels__:
+            opts += [
+                  ['EncryptionPolicySuiteB', [1L] + 3 * [0], '4' + MQLONG_TYPE],
+                  ['CertificateValPolicy', 0L, MQLONG_TYPE]
+                ]
+            
+            opts += [['pad','', '4s']]
 
         apply(MQOpts.__init__, (self, tuple(opts)), kw)
 
