--- outbox.py.orig	2014-11-07 21:46:13.315547611 +0100
+++ outbox.py	2014-11-07 21:46:32.071640619 +0100
@@ -171,7 +171,7 @@
     def disconnect(self):
         self._conn.quit()
 
-    def send(self, email, attachments=()):
+    def send(self, email, attachments=(), from_=None):
         '''Send an email. Connect/Disconnect if not already connected
 
         Arguments:
@@ -185,11 +185,11 @@
             msg['From'] = self.sender_address()
 
         if self._conn:
-            self._conn.sendmail(self.username, email.recipients,
+            self._conn.sendmail(from_ or self.username, email.recipients,
                                 msg.as_string())
         else:
             with self:
-                self._conn.sendmail(self.username, email.recipients,
+                self._conn.sendmail(from_ or self.username, email.recipients,
                                     msg.as_string())
 
     def sender_address(self):
