--- outbox.py.orig	2014-11-07 22:57:21.028710053 +0100
+++ outbox.py	2014-11-07 23:01:49.014038924 +0100
@@ -184,12 +184,25 @@
         if 'From' not in msg:
             msg['From'] = self.sender_address()
 
+        cc = msg.get('CC', [])
+        bcc = msg.get('BCC', [])
+
+        if isinstance(cc, basestring):
+            cc = [cc]
+
+        if isinstance(bcc, basestring):
+            bcc = [bcc]
+
+        recipients = email.recipients
+        recipients.extend(cc)
+        recipients.extend(bcc)
+
         if self._conn:
-            self._conn.sendmail(from_ or self.username, email.recipients,
+            self._conn.sendmail(from_ or self.username, recipients,
                                 msg.as_string())
         else:
             with self:
-                self._conn.sendmail(from_ or self.username, email.recipients,
+                self._conn.sendmail(from_ or self.username, recipients,
                                     msg.as_string())
 
     def sender_address(self):
