--- outbox.py.orig	2014-07-30 23:11:40.932764831 +0200
+++ outbox.py	2014-07-30 23:11:47.028764585 +0200
@@ -5,7 +5,7 @@
 '''
 
 import smtplib
-import sys
+import socket, sys
 
 from email import encoders
 from email.header import Header
@@ -123,13 +123,13 @@
 class Outbox(object):
     '''Thin wrapper around the SMTP and SMTP_SSL classes from the smtplib module.'''
 
-    def __init__(self, username, password, server, port, mode='TLS', debug=False):
+    def __init__(self, username, password, server, port, mode='TLS', debug=False, timeout=socket._GLOBAL_DEFAULT_TIMEOUT):
         if mode not in ('SSL', 'TLS', None):
             raise ValueError("Mode must be one of TLS, SSL, or None")
 
         self.username = username
         self.password = password
-        self.connection_details = (server, port, mode, debug)
+        self.connection_details = (server, port, mode, debug, timeout)
         self._conn = None
 
     def __enter__(self):
@@ -144,14 +144,14 @@
 
         Returns an authenticated SMTP instance.
         '''
-        server, port, mode, debug = self.connection_details
+        server, port, mode, debug, timeout = self.connection_details
 
         if mode == 'SSL':
             smtp_class = smtplib.SMTP_SSL
         else:
             smtp_class = smtplib.SMTP
 
-        smtp = smtp_class(server, port)
+        smtp = smtp_class(server, port, timeout=timeout)
         smtp.set_debuglevel(debug)
 
         if mode == 'TLS':
@@ -227,6 +227,7 @@
 
     part = MIMEBase('application', 'octet-stream')
     part.set_payload(data)
+    part.set_charset('utf-8')
     encoders.encode_base64(part)
     filename = attachment.name if rfc2231 else Header(attachment.name).encode()
     part.add_header('Content-Disposition', 'attachment',
