--- outbox.py.orig	2015-01-06 23:34:15.679566655 +0100
+++ outbox.py	2015-01-06 23:34:23.795566786 +0100
@@ -241,7 +241,6 @@
     part = MIMEBase('application', 'octet-stream')
     part.set_payload(data)
     part.set_charset('utf-8')
-    encoders.encode_base64(part)
     filename = attachment.name if rfc2231 else Header(attachment.name).encode()
     part.add_header('Content-Disposition', 'attachment',
                     filename=filename)
