--- imbox/__init__.py.orig	2015-01-06 21:52:12.795467627 +0100
+++ imbox/__init__.py	2015-01-06 21:52:18.783467723 +0100
@@ -36,7 +36,7 @@
             yield (uid, self.fetch_by_uid(uid))
 
     def mark_seen(self, uid):
-        self.connection.uid('STORE', uid, '+FLAGS', '\\Seen')
+        self.connection.uid('STORE', uid, '+FLAGS', '(\\Seen)')
 
     def delete(self, uid):
         mov, data = self.connection.uid('STORE', uid, '+FLAGS', '(\\Deleted)')
