--- gunicorn/arbiter.py.orig	2019-02-12 11:07:30.253916939 +0100
+++ gunicorn/arbiter.py	2019-02-12 11:07:40.997917112 +0100
@@ -410,7 +410,8 @@
         util._setproctitle("master [%s]" % self.proc_name)
 
         # spawn new workers
-        for i in range(self.cfg.workers):
+        for idx, i in enumerate(range(self.cfg.workers)):
+            os.environ['ZATO_SERVER_WORKER_IDX'] = str(idx)
             self.spawn_worker()
 
         # manage workers
@@ -525,7 +526,8 @@
         of the master process.
         """
 
-        for i in range(self.num_workers - len(self.WORKERS.keys())):
+        for idx, i in enumerate(range(self.num_workers - len(self.WORKERS.keys()))):
+            os.environ['ZATO_SERVER_WORKER_IDX'] = str(idx)
             self.spawn_worker()
             time.sleep(0.1 * random.random())
 
@@ -545,6 +547,8 @@
         :attr sig: `signal.SIG*` value
          """
         try:
+            worker = self.WORKERS[pid]
+            self.cfg.worker_exit(self, worker)
             os.kill(pid, sig)
         except OSError as e:
             if e.errno == errno.ESRCH:
