--- gunicorn/arbiter.py.orig	2016-07-30 12:01:28.691556648 +0200
+++ gunicorn/arbiter.py	2016-07-30 12:01:40.571556840 +0200
@@ -410,7 +410,8 @@
         util._setproctitle("master [%s]" % self.proc_name)
 
         # spawn new workers
-        for i in range(self.cfg.workers):
+        for idx, i in enumerate(range(self.cfg.workers)):
+            os.environ['ZATO_SERVER_WORKER_IDX'] = str(idx)
             self.spawn_worker()
 
         # manage workers
@@ -525,7 +526,8 @@
         of the master process.
         """
 
-        for i in range(self.num_workers - len(self.WORKERS.keys())):
+        for idx, i in enumerate(range(self.num_workers - len(self.WORKERS.keys()))):
+            os.environ['ZATO_SERVER_WORKER_IDX'] = str(idx)
             self.spawn_worker()
             time.sleep(0.1 * random.random())
 
