--- gunicorn/arbiter.py.orig	2019-02-12 16:34:00.990233788 +0100
+++ gunicorn/arbiter.py	2019-02-12 16:34:58.418234717 +0100
@@ -477,7 +477,8 @@
         util._setproctitle("master [%s]" % self.proc_name)

         # spawn new workers
-        for _ in range(self.cfg.workers):
+        for idx, _ in enumerate(range(self.cfg.workers)):
+            os.environ['ZATO_SERVER_WORKER_IDX'] = str(idx)
             self.spawn_worker()

         # manage workers
@@ -612,7 +613,8 @@
         of the master process.
         """

-        for _ in range(self.num_workers - len(self.WORKERS.keys())):
+        for idx, _ in enumerate(range(self.num_workers - len(self.WORKERS.keys()))):
+            os.environ['ZATO_SERVER_WORKER_IDX'] = str(idx)
             self.spawn_worker()
             time.sleep(0.1 * random.random())

