--- gunicorn/glogging.py.orig	2019-02-14 22:57:59.976563959 +0100
+++ gunicorn/glogging.py	2019-02-14 22:58:26.104563776 +0100
@@ -56,11 +56,11 @@
 
         loggers={
             "root": {"level": "INFO", "handlers": ["console"]},
-            "gunicorn.error": {
+            "gunicorn.main": {
                 "level": "INFO",
                 "handlers": ["error_console"],
                 "propagate": True,
-                "qualname": "gunicorn.error"
+                "qualname": "gunicorn.main"
             },
 
             "gunicorn.access": {
@@ -188,7 +188,7 @@
     atoms_wrapper_class = SafeAtoms
 
     def __init__(self, cfg):
-        self.error_log = logging.getLogger("gunicorn.error")
+        self.error_log = logging.getLogger("gunicorn.main")
         self.error_log.propagate = False
         self.access_log = logging.getLogger("gunicorn.access")
         self.access_log.propagate = False
@@ -204,7 +204,7 @@
         self.error_log.setLevel(self.loglevel)
         self.access_log.setLevel(logging.INFO)
 
-        # set gunicorn.error handler
+        # set gunicorn.main handler
         if self.cfg.capture_output and cfg.errorlog != "-":
             for stream in sys.stdout, sys.stderr:
                 stream.flush()
