--- gunicorn/glogging.py.orig	2015-01-16 09:24:13.121546744 +0100
+++ gunicorn/glogging.py	2015-01-16 09:24:33.073547066 +0100
@@ -50,11 +50,11 @@
 
         loggers={
             "root": {"level": "INFO", "handlers": ["console"]},
-            "gunicorn.error": {
+            "gunicorn.main": {
                 "level": "INFO",
                 "handlers": ["console"],
                 "propagate": True,
-                "qualname": "gunicorn.error"
+                "qualname": "gunicorn.main"
             }
         },
         handlers={
@@ -155,7 +155,7 @@
     atoms_wrapper_class = SafeAtoms
 
     def __init__(self, cfg):
-        self.error_log = logging.getLogger("gunicorn.error")
+        self.error_log = logging.getLogger("gunicorn.main")
         self.access_log = logging.getLogger("gunicorn.access")
         self.error_handlers = []
         self.access_handlers = []
