--- apscheduler/triggers/interval.py.orig	2012-08-15 22:48:43.000000000 +0200
+++ apscheduler/triggers/interval.py	2012-08-15 22:52:20.826523665 +0200
@@ -18,7 +18,7 @@
             self.interval_length = 1
 
         if start_date is None:
-            self.start_date = datetime.now() + self.interval
+            self.start_date = datetime.utcnow() + self.interval
         else:
             self.start_date = convert_to_datetime(start_date)
 
