diff --git gunicorn/workers/ggevent.py gunicorn/workers/ggevent.py
index 0c00f53..e99a79e 100644
--- gunicorn/workers/ggevent.py
+++ gunicorn/workers/ggevent.py
@@ -86,8 +86,7 @@ class GeventWorker(AsyncWorker):
         ssl_args = {}
 
         if self.cfg.is_ssl:
-            ssl_args = dict(server_side=True,
-                    do_handshake_on_connect=False, **self.cfg.ssl_options)
+            ssl_args = dict(server_side=True, **self.cfg.ssl_options)
 
         for s in self.sockets:
             s.setblocking(1)