diff --git gunicorn/workers/sync.py gunicorn/workers/sync.py
index b57f28b..40af0e3 100644
--- gunicorn/workers/sync.py
+++ gunicorn/workers/sync.py
@@ -85,8 +85,7 @@ class SyncWorker(base.Worker):
         try:
             if self.cfg.is_ssl:
                 client = ssl.wrap_socket(client, server_side=True,
-                        do_handshake_on_connect=False,
-                        **self.cfg.ssl_options)
+                    **self.cfg.ssl_options)
 
             parser = http.RequestParser(self.cfg, client)
             req = six.next(parser)