--- jsonpointer.py.orig	2014-05-19 21:59:33.909591712 +0200
+++ jsonpointer.py	2014-05-19 22:02:42.225584126 +0200
@@ -60,6 +60,11 @@
 class JsonPointerException(Exception):
     pass
 
+class PathNotFoundException(JsonPointerException):
+    pass
+
+class OutOfBoundsException(JsonPointerException):
+    pass
 
 class EndOfList(object):
     """ Result of accessing element "-" of a list """
@@ -236,7 +241,7 @@
                 return doc[part]
 
             except IndexError:
-                raise JsonPointerException("index '%s' is out of bounds" % (part, ))
+                raise OutOfBoundsException("index '%s' is out of bounds" % (part, ))
 
         else:
             # Object supports __getitem__, assume custom indexing
