--- hvac/v1/__init__.py.orig	2016-11-14 18:56:09.134646418 +0100
+++ hvac/v1/__init__.py	2016-11-14 18:56:35.810646849 +0100
@@ -891,18 +891,18 @@
         self.session.close()
 
     def _get(self, url, **kwargs):
-        return self.__request('get', url, **kwargs)
+        return self._request('get', url, **kwargs)
 
     def _post(self, url, **kwargs):
-        return self.__request('post', url, **kwargs)
+        return self._request('post', url, **kwargs)
 
     def _put(self, url, **kwargs):
-        return self.__request('put', url, **kwargs)
+        return self._request('put', url, **kwargs)
 
     def _delete(self, url, **kwargs):
-        return self.__request('delete', url, **kwargs)
+        return self._request('delete', url, **kwargs)
 
-    def __request(self, method, url, headers=None, **kwargs):
+    def _request(self, method, url, headers=None, **kwargs):
         url = urljoin(self._url, url)
 
         if not headers:
@@ -933,11 +933,11 @@
                 errors = response.json().get('errors')
             if errors is None:
                 text = response.text
-            self.__raise_error(response.status_code, text, errors=errors)
+            self._raise_error(response.status_code, text, errors=errors)
 
         return response
 
-    def __raise_error(self, status_code, message=None, errors=None):
+    def _raise_error(self, status_code, message=None, errors=None):
         if status_code == 400:
             raise exceptions.InvalidRequest(message, errors=errors)
         elif status_code == 401:
