--- ws4py/server/geventserver.py.orig	2016-08-07 03:15:14.379510013 +0200
+++ ws4py/server/geventserver.py	2016-08-07 03:15:28.407510240 +0200
@@ -73,12 +73,10 @@
     """
 
     def track(self, websocket):
-        logger.info("Managing websocket %s" % format_addresses(websocket))
         return self.spawn(websocket.run)
 
     def clear(self):
-        logger.info("Terminating server and all connected websockets")
-        for greenlet in self:
+        for greenlet in list(self):
             try:
                 websocket = greenlet._run.im_self
                 if websocket:
