--- sqlalchemy/sql/crud.py.orig	2020-01-11 20:53:25.718517840 +0100
+++ sqlalchemy/sql/crud.py	2020-01-11 20:53:31.326681354 +0100
@@ -165,18 +165,6 @@
             kw,
         )
 
-    if parameters and stmt_parameters:
-        check = (
-            set(parameters)
-            .intersection(_column_as_key(k) for k in stmt_parameters)
-            .difference(check_columns)
-        )
-        if check:
-            raise exc.CompileError(
-                "Unconsumed column names: %s"
-                % (", ".join("%s" % c for c in check))
-            )
-
     if stmt._has_multi_parameters:
         values = _extend_values_for_multiparams(compiler, stmt, values, kw)
 
