--- redis/connection.py.orig	2016-03-23 01:22:36.921599519 +0100
+++ redis/connection.py	2016-03-23 01:22:45.817599663 +0100
@@ -8,6 +8,9 @@
 import threading
 import warnings
 
+# Added for Zato
+from logging import DEBUG, getLogger
+
 try:
     import ssl
     ssl_available = True
@@ -414,6 +417,8 @@
             'port': self.port,
             'db': self.db,
         }
+        self.logger = getLogger('zato_kvdb')
+        self.has_debug_log = self.logger.isEnabledFor(DEBUG)
         self._connect_callbacks = []
 
     def __repr__(self):
@@ -560,6 +565,8 @@
 
     def send_command(self, *args):
         "Pack and send a command to the Redis server"
+        if self.has_debug_log:
+            self.logger.debug('Cmd: `%s:%s (db:%s)`, `%r`', self.host, self.port, self.db, args)
         self.send_packed_command(self.pack_command(*args))
 
     def can_read(self, timeout=0):
