--- redis/connection.py.orig	2014-05-21 13:40:28.124505304 +0200
+++ redis/connection.py	2014-05-21 16:39:15.661482188 +0200
@@ -5,6 +5,8 @@
 import sys
 import threading
 
+# Added for Zato
+from logging import getLogger
 
 from redis._compat import (b, xrange, imap, byte_to_chr, unicode, bytes, long,
                            BytesIO, nativestr, basestring,
@@ -232,6 +234,7 @@
             'port': self.port,
             'db': self.db,
         }
+        self.logger = getLogger('zato_kvdb')
 
     def __repr__(self):
         return self.description_format % self._description_args
@@ -331,6 +334,7 @@
 
     def send_command(self, *args):
         "Pack and send a command to the Redis server"
+        self.logger.debug('Cmd: `%s:%s (db:%s)`, `%r`', self.host, self.port, self.db, args)
         self.send_packed_command(self.pack_command(*args))
 
     def read_response(self):
